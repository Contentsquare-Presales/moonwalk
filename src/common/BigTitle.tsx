import React from 'react';
import styled from 'styled-components';
import { useTheme } from '@react-navigation/native';

const TitleText = styled.Text`
  font-size: 26px;
  font-weight: bold;
  margin: 10px 20px;
  font-family: Quicksand;
`;

const BigTitle = ({title}: {title: string}) => {

  const {colors} = useTheme();

  return <TitleText style={{color: colors.text}}>{title}</TitleText>
}

export default BigTitle;