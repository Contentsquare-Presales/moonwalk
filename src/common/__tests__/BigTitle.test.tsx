import React from "react";
import BigTitle from "../BigTitle";
import renderer from "react-test-renderer";

describe("BigTitle", () => {
  it("renders correctly", () => {
    const props = {
      title: "$_TITLE_TEXT_$",
    };
    const tree = renderer.create(<BigTitle {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with the See More option", () => {
    const props = {
      title: "$_TITLE_TEXT_$",
      onSeeMore: jest.fn(),
    };
    const tree = renderer.create(<BigTitle {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});