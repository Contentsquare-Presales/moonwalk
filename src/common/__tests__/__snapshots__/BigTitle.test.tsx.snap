// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigTitle renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    $_TITLE_TEXT_$
  </Text>
</View>
`;

exports[`BigTitle renders correctly with the See More option 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    $_TITLE_TEXT_$
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {},
          Object {
            "color": "rgb(28, 28, 30)",
          },
        ]
      }
    >
      See more
    </Text>
  </View>
</View>
`;
